packages/manager/cypress/e2e/firewalls/update-firewall.spec.ts

```diff
@@ -0,0 +1,337 @@
+ *
+ * @param rule - the firewall rule to be added.
+ * @param direction - the direction of the rule, inbound or outbound.
+ */
+const addFirewallRules = (rule: FirewallRuleType, direction: string) => {
+  const ruleType =
+    direction && direction.toLowerCase() === 'outbound'
+      ? 'Outbound'
+      : 'Inbound';
+
+  // Go to Rules tab
+  ui.tabList.findTabByTitle('Rules').should('be.visible').click();
+
+  ui.button.findByText(ruleType).should('be.visible').click();
+
+  ui.drawer
+    .findByTitle(`Add an ${ruleType} Rule`)
+    .should('be.visible')
+    .within(() => {
+      const port = rule.ports ? rule.ports : '22';
+      cy.get('[data-qa-enhanced-select="Select a rule preset..."]').type(
+        portPresetMap[port] + '{enter}'
+      );
+
+      const label = rule.label ? rule.label : 'test-label';
+      const description = rule.description
+        ? rule.description
+        : 'test-description';
+      containsClick('Label').type('{selectall}{backspace}' + label);
+      containsClick('Description').type(description);
+
+      const action = rule.action
+        ? rule.action.charAt(0).toUpperCase() +
+          rule.action.slice(1).toLowerCase()
+        : 'Accept';
+      containsClick(action).click();
+      ui.button.findByText('Add Rule').should('be.visible').click();
+    });
+};
+
+/**
+ * Removes an inboun / outbount firewall rule.
+ *
+ * No assertion is made on the result of the rule addition attempt.
+ *
+ * @param ruleLabel - the label of the firewall rule to be removed.
+ */
+const removeFirewallRules = (ruleLabel: string) => {
+  // Go to Rules tab
+  ui.tabList.findTabByTitle('Rules').should('be.visible').click();
+
+  cy.get(`[aria-label="${ruleLabel}"]`)
+    .first()
+    .should('be.visible')
+    .within(() => {
+      ui.button.findByText('Delete').should('be.visible').click();
+    });
+};
+
+/**
+ * Adds a linode to the firewall.
+ *
+ * No assertion is made on the result of the rule addition attempt.
+ *
+ * @param firewall - the firewall that is modified.
+ * @param linode - the linode that is added to the firewall.
+ */
+const addLinodesToFirewall = (firewall: Firewall, linode: Linode) => {
+  // Go to Linodes tab
+  ui.tabList.findTabByTitle('Linodes').should('be.visible').click();
+
+  cy.get('button')
+    .contains('span', 'Add Linodes to Firewall')
+    .should('be.visible')
+    .click();
+
+  ui.drawer
+    .findByTitle(`Add Linode to Firewall: ${firewall.label}`)
+    .should('be.visible')
+    .within(() => {
+      // Fill out and submit firewall edit form.
+      cy.get('[data-testid="textfield-input"]:last')
+        .should('be.visible')
+        .click()
+        .type(linode.label);
+    });
+
+  ui.autocompletePopper.findByTitle(linode.label).should('be.visible').click();
+
+  ui.drawer
+    .findByTitle(`Add Linode to Firewall: ${firewall.label}`)
+    .should('be.visible')
+    .within(() => {
+      ui.button.findByText('Add').click();
+    });
```

Comment:
```suggestion
  ui.drawer
    .findByTitle(`Add Linode to Firewall: ${firewall.label}`)
    .should('be.visible')
    .within(() => {
      // Fill out and submit firewall edit form.
      cy.get('[data-testid="textfield-input"]:last')
        .should('be.visible')
        .click()
        .type(linode.label);
        
      ui.autocompletePopper
        .findByTitle(linode.label)
        .should('be.visible')
        .click();
          
      ui.button.findByText('Add').click();
    });
```

I think we can simplify this so that it's all inside one `within(() => { ... })` block, since the `autocompletePopper` UI helper breaks out of the `within(() => { ... })` context.